Behaviour Change for `_count` Operator - RN (#4438)
SumoLogic/sumologic-documentation: blog-service/2024-08-21-search.md

---

> title: Behaviour Change for `count` Operator (Search)

Reviewer: ```suggestion
title: Behavior Change for `count` Search Operator (Search)
```

image: https://help.sumologic.com/img/sumo-square.png
keywords:
  - count
  - search-operator
  - log-search
hide_table_of_contents: true
authors:
  - url: https://help.sumologic.com/release-notes-service/rss.xml
    image_url: /img/release-notes/rss-orange.png
---

> We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values. 

Reviewer 2: a change to the behaviour of the `count` operator to return `Long` values.  Previously, the `count` operator returned `Int` data type which would result in a negative value when exceeding the maximum integer value. With this change, we are changing  the data type of the `count` operator result from `Int` to `Long` data type.

> We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values. 
> 
> This change will apply to all new queries. Any existing queries as part of various content types - Log Search, Scheduled Search, and Saved Search will also seamlessly transition to show `Long` values in case of `Int` overflow.

Reviewer: ```suggestion
We're happy to announce an update to the behavior of our log search `count` operator. Previously, the `count` operator used the `Int` data type, which could lead to negative `count` values if the maximum integer value was exceeded. To address this, we have changed the data type of the `count` operator from `Int` to `Long`, allowing for the handling of larger numeric values without overflow.

This change will apply to all new queries. Existing queries in Log Search, Scheduled Search, and Saved Search will automatically transition to show `Long` values in cases where an `Int` overflow would have occurred.
```

> This change will apply to all new queries. Any existing queries as part of various content types - Log Search, [Scheduled Search](/docs/manage/scheduled-views/scheduled-views-best-practices/#upgrade-your-scheduled-views-to-avoid-negative-count-results), [Lookup Table](/docs/search/lookup-tables/manage-update-lookup-tables/#upgrade-the-lookup-table-to-avoid-negative-count-results), and Saved Search will also seamlessly transition to show `Long` values in case of `Int` overflow.

Reviewer 2: hyperlink is for Scheduled View. so please add Scheduled View after scheduled Search and link this particular URL with scheduled view.

---

> title: Behaviour Change for `_count` Operator (Search)

Reviewer 3: ```suggestion
title: Behaviour Change for `count` Operator (Search)
```

image: https://help.sumologic.com/img/sumo-square.png
keywords:
  - count
  - search-operator
  - log-search
hide_table_of_contents: true
authors:
  - url: https://help.sumologic.com/release-notes-service/rss.xml
    image_url: /img/release-notes/rss-orange.png
---

> We're happy to announce a change in the behavior of the `count` operator. Previously, the `count` operator was supported by the `Int` data type, which could result in a negative count value after when exceeding the maximum integer value. To address this, we are changing the data type of the `count` operator from `Int` to `Long` data type to support the handling of large numeric values. 

Reviewer 2: "...which could result in a negative count value ~after~ when exceeding the..."

This change will apply to all new queries. However, it may pose challenges for existing saved searches, scheduled searches, searches using the `count` operator as part of the scheduled view, and searches using the `count` operator as part of lookup tables. Below are some conditions to address these challenges:

> - For existing scheduled views, the query will use the `Int` data type, while new scheduled views will utilize the `Long` data type for the `count` operator.

Reviewer 2: the scheduled view related callout should be added to the Scheduled Views page with a callout on the release note for users to expect newer scheduled views to support Long as the data type for count operator

> - For existing lookup tables, lookup schema matching has been relaxed to allow Int-Long type mismatch, in order to support this new behavioural change. If the schema type is `Int` and the field type is `Long`, there can be possible overflow, resulting in a negative count value after crossing the maximum integer value. If the schema type is `Long` and the field type is `Int` there will be no issues.

Reviewer 2: the lookup related callout should be added to the lookup page with a callout on the release note for users to expect newer Lookup to only support Long as the data type for count operator
